tests/unit/test_contrib/test_sqlalchemy/test_repository/conftest.py

```diff
@@ -57,7 +57,7 @@ def fx_raw_authors_uuid() -> list[dict[str, Any]]:
             "updated_at": "2023-05-11T00:00:00",
         },
         {
-            "id": "5ef29f3c-3560-4d15-ba6b-a2e5c721e4d2",
+            "id": UUID("5ef29f3c-3560-4d15-ba6b-a2e5c721e4d2"),
```

Comment:
In this case, I left it stringified to verify that it would be properly cast to UUID.

Reply:
Maybe it should be its own test case though?

Reply:
Yeah, that's what I assumed. There's a few things that need to be cleaned up still, I just wanted to see if I could get it working at all first :D
I just did this so it's easier to handle the case where we compare the PKs directly.

Should probably be its own test, yes.

Reply:
The same test should be done for the BigInt model I think, where a string is cast to an int?

Reply:
Okay, looks like I need to be more selective about this. There are a few failures which I *think* are correct, since we don't explicitly handle those cases (e.g. passing in a `str` as a PK a bigint model). 

I guess the question is if we *do* want to handle those cases, or if we leave it up to the user to provide a PK value with an appropriate type?

Reply:
Shall we spin that off into a separate PR then?

Reply:
As for the `str` > `int` conversion: If we simply handle this as a method on the repo (e.g. some method that depends on the type declared for the id column), then supporting this alongside UUIDs would basically be zero effort, so unless there's a reason not do to this, I think we might as well support it. Wdyt?

Reply:
I like this idea